File: bug17_invalid_list_index/main.java
import java.nio.ByteBuffer;
import java.util.Base64;

public class main {
    public static void main(String[] args) {
        long[] arr = {0xA1B2L, 0xC3D4L, 0xE5F6L, 0x0718L, 0x293AL, 0x4B5CL, 0x6D7EL, 0x8F90L, 0xA1B3L, 0xC4D5L};
        long acc = 0;
        for (int step = 0; step < 10; step++) {
            int idx = step * 3 % 11;
            if (idx >= 10 || idx < 0) {
                throw new IndexOutOfBoundsException("IndexError: list index out of range");
            }
            acc ^= arr[idx];
            acc *= 0x9E3779B97F4A7C15L;
            acc = (acc << 13) | (acc >>> 51);
        }
        ByteBuffer buffer = ByteBuffer.allocate(8);
        buffer.putLong(acc);
        System.out.println(Base64.getEncoder().encodeToString(buffer.array()));
    }
}

File: bug17_invalid_list_index/main.js
function run() {
    let arr = [0xA1B2n, 0xC3D4n, 0xE5F6n, 0x0718n, 0x293An, 0x4B5Cn, 0x6D7En, 0x8F90n, 0xA1B3n, 0xC4D5n];
    let acc = 0n;
    for (let step = 0n; step < 10n; step++) {
        let idx = Number((step * 3n) % 11n);
        if (idx >= 10 || idx < 0) {
            throw new Error("IndexError: list index out of range");
        }
        acc ^= arr[idx];
        acc = BigInt.asUintN(64, acc * 0x9E3779B97F4A7C15n);
        acc = BigInt.asUintN(64, (acc << 13n) | (acc >> 51n));
    }
    let buf = Buffer.alloc(8);
    buf.writeBigUInt64BE(acc, 0);
    console.log(buf.toString('base64'));
}
run();
